manticore/core/memory.py

```diff
@@ -771,13 +772,54 @@ def read(self, addr, size):
 
         return result
 
+
+    def start_write_trace(self):
+        '''
+        Begin recording all writes. Retrieve all writes with `stop_write_trace()`
+
+        :return: None
+        '''
+        print '[d] starting trace..'
+        self._recording_stack.append([])
+
+    def stop_write_trace(self):
+        '''
+        Stop recording trace and return a `list[(address, value)]` of all the writes
+        that occurred, where `value` is of type list[str]. Can be called without
```

Comment:
Nope, I added trace support to `Memory.write`; `SMemory.write` records a list of symbols separately and only falls through to `super` if the value is concrete. ( https://github.com/trailofbits/manticore/blob/baa318e308f48b2205e3edbe5aa005025bf4dd18/manticore/core/memory.py#L1019 )